pkg/registry/core/pod/strategy.go

```diff
@@ -205,6 +205,25 @@ type podEphemeralContainersStrategy struct {
 // EphemeralContainersStrategy wraps and exports the used podStrategy for the storage package.
 var EphemeralContainersStrategy = podEphemeralContainersStrategy{Strategy}
 
+// dropNonEphemeralContainerUpdates discards all changes except for pod.Spec.EphemeralContainers and certain metadata
+func dropNonEphemeralContainerUpdates(newPod, oldPod *api.Pod) {
```

Comment:
one deep copy is better, can return `pod` and overwrite `newPod` at the callsite

```
func dropNonEphemeralContainerUpdates(newPod, oldPod *api.Pod) *api.Pod {
  pod := oldPod.DeepCopy()
  pod.Name = newPod.Name
  ...
  return pod
}
```

Reply:
SG, done.